lang/codegen/src/generator/module_impl/deploy.rs

```diff
@@ -93,12 +93,66 @@ where
     args.into_iter()
         .map(|arg| {
             let pat = &*arg.pat;
-            quote!(args
-                .get(stringify!(#pat))
-                .cloned()
-                .unwrap()
-                .into_t()
-                .unwrap())
+            match &*arg.ty {
+                syn::Type::Reference(ty) => match &*ty.elem {
+                    syn::Type::Array(inner_ty) => {
+                        let inner_ty = &inner_ty.elem;
+                        quote! {
```

Comment:
You could extract functions. There are 2 repeating quotes - with a plain type and a Vec.